pkg/docker/creds/credentials_test.go

```diff
@@ -170,7 +169,30 @@ func TestCheckAuth(t *testing.T) {
 				Username: tt.args.username,
 				Password: tt.args.password,
 			}
-			if err := creds.CheckAuth(tt.args.ctx, tt.args.registry+"/someorg/someimage:sometag", c, http.DefaultTransport); (err != nil) != tt.wantErr {
+			// create trusted certificates pool and add our certificate
+			certPool := x509.NewCertPool()
+			certPool.AddCert(cert)
+
+			// client transport with the certificate
+			transport := &http.Transport{
```

Comment:
the `startServer` whichi previously used to handle the http transport mutation gets used by one more function `TestCheckAuthEmptyCreds` which still uses the default http transport instead of this one but doesnt seem to cause any problems? 
Perhaps `TestCheckAuthEmptyCreds` needs to have the Transport changed as well